airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py

```diff
@@ -118,10 +131,18 @@ def __init__(
         self.dynamodb_scan_kwargs = dynamodb_scan_kwargs
         self.s3_bucket_name = s3_bucket_name
         self.s3_key_prefix = s3_key_prefix
-        self.aws_conn_id = aws_conn_id
+        if aws_conn_id is not NOTSET:
+            warnings.warn(_DEPRECATION_MSG, DeprecationWarning, stacklevel=3)
+            self.source_aws_conn_id = aws_conn_id
+        else:
+            self.source_aws_conn_id = source_aws_conn_id
```

Comment:
Should there be an exception if `aws_conn_id` and `source_aws_conn_id` are both provided?  If not, I think it should be made pretty obvious which has priority.  Maybe the fact you are deprecating `aws_conn_id` is enough of a hint, but maybe not.  Thoughts?

Reply:
I personally feel that if you're getting a warning about deprecation when you use both, it should be helpful enough and we do want people to migrate away from `aws_conn_id` eventually.